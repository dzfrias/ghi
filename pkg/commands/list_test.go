package commands

import (
	"os"
	"testing"

	"github.com/dzfrias/ghi/pkg/issues"
	"github.com/dzfrias/ghi/pkg/testutil"
	"github.com/stretchr/testify/assert"
)

var data = issues.IssuesSearchResult{
	TotalCount: 1,
	Items: []*issues.Issue{
		{
			Number:  1,
			HTMLURL: "none",
			Title:   "Testing",
			State:   "open",
			User: &issues.User{
				Login:   "TestUser",
				HTMLURL: "none",
			},
		},
	},
}

var app = testutil.NewApp(List)

func TestList(t *testing.T) {
	got := testutil.CapStdout(&out, func() {
		args := os.Args[0:1]
		args = append(args, "list", "repo")
		err := app.Run(args)
		assert.Nil(t, err)
	})
	target := "1 issues:\n#1      TestUser Testing\n"
	assert.Equal(t, target, got)
}

func TestListBadPage(t *testing.T) {
	args := os.Args[0:1]
	args = append(args, "list", "--page", "0", "repo")
	err := app.Run(args)
	assert.Error(t, err)
}

func TestMain(m *testing.M) {
	server := testutil.SetupIssueServer(data)
	defer server.Close()
	issues.IssuesURL = server.URL
	os.Exit(m.Run())
}
