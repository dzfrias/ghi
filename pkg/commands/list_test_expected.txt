21 issues:
#1      TestUser Testing
#2      TestUser Testing
#3      TestUser Testing
#4      TestUser Testing
#5      TestUser Testing
#6      TestUser Testing
#7      TestUser Testing
#8      TestUser Testing
#9      TestUser Testing
#10     TestUser Testing
#11     TestUser Testing
#12     TestUser Testing
#13     TestUser Testing
#14     TestUser Testing
#15     TestUser Testing
#16     TestUser Testing
#17     TestUser Testing
#18     TestUser Testing
#19     TestUser Testing
#20     TestUser Testing
(Showing issues 1-20)
